### PR TITLE
chore: publish symbols internally

### PR DESCRIPTION
**Pull request by rzhao271**
This PR adds PublishSymbols tasks to publish CLI and VS Code symbols internally.
I was not able to get `IndexSources: true` to work. Setting IndexSources to true resulted in errors about ADO being unable to index the sources because the sources are on GitHub.

Verification build: https://dev.azure.com/monacotools/Monaco/_build/results?buildId=300012&view=results

### CODE REVIEW

File: build/azure-pipelines/cli/cli-compile.yml [hunk 1]
```diff
@@ -108,6 +108,14 @@ steps:
             ${{ pair.key }}: ${{ pair.value }}
 
     - ${{ if contains(parameters.VSCODE_CLI_TARGET, '-windows-') }}:
+      - task: PublishSymbols@2
+        inputs:
+          IndexSources: false
+          SymbolsFolder: $(Build.SourcesDirectory)/cli/target/${{ parameters.VSCODE_CLI_TARGET }}/release
+          SearchPattern: 'code.pdb'
+          SymbolServerType: TeamServices
```
Comment (karthiknadig) [inline build/azure-pipelines/cli/cli-compile.yml:116]:
```suggestion
          SymbolServerType: TeamServices
          SymbolsProduct: 'code'
```

File: build/azure-pipelines/win32/sdl-scan-win32.yml [hunk 2]
```diff
@@ -139,6 +139,15 @@ steps:
       flattenFolders: true
     condition: succeeded()
 
+  - task: PublishSymbols@2
+    inputs:
+      IndexSources: false
+      SymbolsFolder: '$(Agent.BuildDirectory)\symbols'
+      SearchPattern: '**\*.pdb'
+      SymbolServerType: TeamServices
```
Comment (karthiknadig) [inline build/azure-pipelines/win32/sdl-scan-win32.yml:147]:
```suggestion
      SymbolServerType: TeamServices
      SymbolsProduct: 'code'
```

**Review by karthiknadig — COMMENTED** [review]
I think symbols might be easier to find if you add product as `code`. Not sure if there is guidance on this.

**Review by karthiknadig — APPROVED** [review]

**Comment by rzhao271** [comment]
I believe the parameter will help, and that we could change it to 'Visual Studio Code' later if it is too vague.

**Pull request merged**